client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -208,13 +208,15 @@ def _build_item_menu_for_selection(self, menu, indexes, active_index):
         filtered_items = []
         product_ids = set()
         version_ids = set()
+        project_names = set()
```

Comment:
You should store product ids by project name and call `get_version_items` per project.